src/cargo/ops/registry/publish.rs

```diff
@@ -705,19 +745,6 @@ fn package_list(pkgs: impl IntoIterator<Item = PackageId>, final_sep: &str) -> S
 }
 
 fn validate_registry(pkgs: &[&Package], reg_or_index: Option<&RegistryOrIndex>) -> CargoResult<()> {
-    let unpublishable = pkgs
```

Comment:
Small nit, feel free to pass on. I'm mentioning this only since I had other comments.

Would make reviewing easier if ĵust moving this error check was a separate refactor commit.